substrate/primitives/runtime/src/traits/transaction_extension/mod.rs

```diff
@@ -0,0 +1,512 @@
+	/// transaction, that can pay for its execution and quickly eliminate ones that are stale or
+	/// incorrect.
+	///
+	/// Parameters:
+	/// - `origin`: The origin of the transaction which this extension inherited; coming from an
+	///   "old-school" *signed transaction*, this will be a system `RawOrigin::Signed` value. If the
+	///   transaction is a "new-school" *General Transaction*, then this will be a system
+	///   `RawOrigin::None` value. If this extension is an item in a composite, then it could be
+	///   anything which was previously returned as an `origin` value in the result of a `validate`
+	///   call.
+	/// - `info`: Information concerning, and inherent to, the transaction's call.
+	/// - `len`: The total length of the encoded transaction.
+	/// - `inherited_implication`: The *implication* which this extension inherits. This is a tuple
+	///   of the transaction's call and some additional opaque-but-encodable data. Coming directly
+	///   from a transaction, the latter is [()]. However, if this extension is expressed as part of
+	///   a composite type, then the latter component is equal to any further implications to which
+	///   the returned `origin` could potentially apply. See Pipelines, Inherited Implications, and
+	///   Authorized Origins for more information.
+	/// - `context`: Some opaque mutable context, as yet unused.
+	///
+	/// Returns a [ValidateResult], which is a [Result] whose success type is a tuple of
+	/// [ValidTransaction] (defining useful metadata for the transaction queue), the [Self::Val]
+	/// token of this transaction, which gets passed into [prepare](TransactionExtension::prepare),
+	/// and the origin of the transaction, which gets passed into
+	/// [prepare](TransactionExtension::prepare) and is ultimately used for dispatch.
+	fn validate(
+		&self,
+		origin: OriginOf<Call>,
+		call: &Call,
+		info: &DispatchInfoOf<Call>,
+		len: usize,
+		context: &mut Context,
+		self_implicit: Self::Implicit,
+		inherited_implication: &impl Encode,
+	) -> ValidateResult<Self::Val, Call>;
+
+	/// Do any pre-flight stuff for a transaction after validation.
+	///
+	/// This is for actions which do not happen in the transaction queue but only immediately prior
+	/// to the point of dispatch on-chain. This should not return an error, since errors should
+	/// already have been identified during the [validate](TransactionExtension::validate) call. If
+	/// an error is returned, the transaction will be considered invalid but no state changes will
+	/// happen and therefore work done in [validate](TransactionExtension::validate) will not be
+	/// paid for.
+	///
+	/// Unlike `validate`, this function may consume `self`.
+	///
+	/// IMPORTANT: **Checks made in validation need not be repeated here.**
+	fn prepare(
+		self,
+		val: Self::Val,
+		origin: &OriginOf<Call>,
+		call: &Call,
+		info: &DispatchInfoOf<Call>,
+		len: usize,
+		context: &Context,
```

Comment:
It's a variable that's supposed to inject any necessary data into the `validate` -> `prepare` flow as well as for `post_dispatch`. For existing `SignedExtension`s migrated to `TransactionExtension`s, the context type is `()` because no such data is needed, but future `TransactionExtension` impls may need this additional bit of information to correctly process the transaction.

Reply:
yes, that's my understanding. I was just suggesting to mention it in the doc, that it is the same instance of the context, there is no info about it now, hence it's not part of the contract.